### ISSUE TITLE
[Kubernetes] Check Kubernetes version before fetching resources

### ISSUE THREAD

**Issue by goober** [backstage/backstage#6109]
## Feature Suggestion

For old clusters that do not have the networking.k8s.io/v1beta1/ingresses resource an error will be shown for all services. See example below:
![image](https://user-images.githubusercontent.com/124244/122557093-d9223180-d02b-11eb-9273-aa517882ec8a.png)

I would like to propose a feature where the kubernetes plugin takes the available resources into account before fetching data from the cluster. And gracefully ignore those resources that are not available.

## Possible Implementation

Backstage could do proper filtering depending what resources are available. `kubectl get-versions` will give you a list of available resources and then use this list to filter out those resources that are not available.

**Comment by benjdlambert** [comment]
@mclarke47 for visibility 👍

**Comment by mclarke47** [comment]
Makes a lot of sense, after we load in the clusters we should compare what resources we intend to fetch with what is configured and not fetch resources we can't get.

**Comment by ananthulasrikar** [comment]
@goober I fixed it using apiGroups: ["networking.k8s.io"] in ClusterRole. Hope it helps.

**Comment by benjdlambert** [comment]
@ananthulasrikar, yeah I think that's a fix for when the groups are hidden because of auth and RBAC, but I think this ticket is more scoped towards those resources / `apiGroups` not existing on everyone's cluster.

**Comment by goober** [comment]
Yes. Without correct RBAC you get 403 in response from the cluster. But as @benjdlambert says the ticket is about when the resource is not available at all, hence the 404

**Comment by tszngai** [comment]
I'm having the same issue. Hope this would get fixed soon

**Comment by mclarke47** [comment]
@dweber019 a workaround would be to use the `kubernetes. objectTypes` config option to specify the types explicitly.

However, I think long term we should consult the api-versions

**Comment by mclarke47** [comment]
Have a look at the solution in https://github.com/backstage/backstage/issues/12095 and let me know what you think

**Comment by mclarke47** [comment]
first step https://github.com/backstage/backstage/pull/13871
